### PR TITLE
Partially fix duplicate entries in RAM module

### PR DESCRIPTION
**Pull request by Yoshi2889**
This is related to #169. While this PR does not fix the issue, it does prevent duplicate processes from appearing in the RAM list due to not having enough processes returned from `top`. 

Should this possibly be applied to the other modules as well?

### CODE REVIEW

**Comment by exelban** [comment]
Hi.

According to this [issue](https://github.com/exelban/stats/issues/169). This bug (duplicating processes) was fixed. So this fix not necessary. But it could be if you change a little bit the moment when cleaning the process view.

Just check if `self.processes` count and `list` count are different. If yes, clean all views.

File: Modules/Battery/popup.swift [hunk 1]
```diff
@@ -254,20 +254,24 @@ internal class Popup: NSView, Popup_p {
...
-                        self.processes[index].icon = process.icon
-                        self.processes[index].toolTip = "pid: \(process.pid)"
-                    }
+            if !(self.window?.isVisible ?? false) && self.processesInitialized {
+                return;
+            }
+            
+            if list.count != self.processes.count {
+                self.processes.forEach { processView in
+                    processView.clear();
```
Comment (exelban) [inline Modules/Battery/popup.swift:263]: Remove semicolons please

**Comment by Yoshi2889** [comment]
Thanks for your feedback! I hope I incorporated it properly in my new commits. Please let me know if things need changing.

I took the liberty of doing a little bit of code cleanup, as there is a lot of duplication going on.

**Comment by exelban** [comment]
Thanks for implementing this in all modules which have top processes.

File: Modules/CPU/popup.swift [hunk 2]
```diff
@@ -236,20 +236,24 @@ internal class Popup: NSView, Popup_p {
...
-                    let process = list[i]
-                    let index = list.count-i-1
-                    if self.processes.indices.contains(index) {
-                        self.processes[index].label = process.name != nil ? process.name! : process.command
-                        self.processes[index].value = "\(process.usage)%"
-                        self.processes[index].icon = process.icon
-                        self.processes[index].toolTip = "pid: \(process.pid)"
-                    }
+            if !(self.window?.isVisible ?? false) && self.processesInitialized {
+                return;
```
Comment (exelban) [inline Modules/CPU/popup.swift:240]: Still a few lefts)

**Comment by Yoshi2889** [comment]
That should be all of them. I need to spend some more time with Swift, haha.

**Comment by exelban** [comment]
Thank you for your contribution. In swift semicolons unnesesary)

**Pull request merged**